design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -0,0 +1,298 @@
+the VM interface should connect to:
+
+```go
+type VirtualMachineInstanceNetworkInterface struct {
+...
+// PodInterfaceName represents the name of the pod network interface
+PodInterfaceName string `json:"podInterfaceName,omitempty"`
+...
+}
+```
+
+VMI controller will set this field's value from the following sources:
+
+- Multus' network-status annotation
+- A local computation
+
+virt-handler and virt-launcher shall read this value instead of independently calculate it.
+
+### API Example
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: myvmi
+spec:
+  ...
+status:
+  ...
+  interfaces:
+      - infoSource: 'domain, guest-agent, multus-status'
+        podInterfaceName: eth0
+        interfaceName: eth0
+        ipAddress: 10.131.0.29
+        ipAddresses:
+          - 10.131.0.29
+        mac: '52:54:00:c0:35:78'
+        name: passtnet
+        queueCount: 1
+  ...
+```
+
+### Pros
+
+- Fully automated
+- No configuration is required in KubeVirt
+- Provides backward compatibility
+- This solution has the potential to ease future changes in the name generation, as it is generated once at the
+  controller
+  and not at each component (which may cause compatibility issues when the system is upgraded).
+
+### Cons
+
+- Adds API
+- Exposes an implementation detail to the VM owner
+- Increases KubeVirt’s coupling with Multus
+- Adds API
+- Will require dedicated tests to verify its functionality
+
+## Alternative 2 - Report Primary Pod Interface Name
+
+Currently, the `VirtualMachineInstanceStatus` struct holds a slice of interface status objects:
+
+```go
+type VirtualMachineInstanceStatus struct {
+...
+Interfaces []VirtualMachineInstanceNetworkInterface `json:"interfaces,omitempty"`
+...
+}
+```
+
+Add new field to report the primary pod interface's name:
+
+```Go
+type VirtualMachineInstanceStatus struct {
+...
+Interfaces []VirtualMachineInstanceNetworkInterface `json:"interfaces,omitempty"`
+
+// PrimaryPodInterfaceName represents the pod primary NIC's name
+PrimaryPodInterfaceName string `json:primaryPodInterfaceName,omitempty`
+...
+}
+```
+
+VMI controller will set this field's value from Multus' network-status annotation, or fallback to `eth0` in case Multus
+is not deployed.
+
+### API Example
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  name: myvmi
+spec:
+  ...
+status:
+  ...
+  primaryPodInterfaceName: eth0
+  interfaces:
+    - infoSource: 'domain, guest-agent, multus-status'
+      interfaceName: eth0
+      ipAddress: 10.131.0.29
+      ipAddresses:
+        - 10.131.0.29
+      mac: '52:54:00:c0:35:78'
+      name: passtnet
+      queueCount: 1
+  ...
+```
+
+### Pros
+
+- Less API than the first alternative
+- Exposes less implementation details to the VM owner.
+- Fully automated
+- No configuration is required in KubeVirt
+- Provides backward compatibility
+
+### Cons
```

Comment:
@EdDev commented:
- The added information is probably not useful for a user.
- The added information is not useful for anything beyond this primary network and it opens a big question: What is the primary network and when it is even relevant.

Reply:
Added these points to the text.
Thank you.